### PR TITLE
Add new protoc-gen-alias command to allow backwards version compat

### PR DESCRIPTION
**Pull request by howardjohn**
Implements https://github.com/istio/api/issues/3127

### CODE REVIEW

File: cmd/protoc-gen-alias/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+`protoc-gen-jsonshim` is a plugin for protoc which generates `MarshalJSON()` and
+`UnmarshalJSON()` functions for `.pb.go` types.  The implementations simply
+redirect to `MarshalJSONPB()` and `UnmarshalJSONPB()` respectively.  This allows
+the types to be used with native Go JSON serialization.
+
+## Usage
+
+Add the executable to your system's PATH, for example:
+
+`$ go install istio.io/tools/cmd/protoc-gen-golang-jsonshim`
```
Comment (keithmattix) [inline cmd/protoc-gen-alias/README.md:12]: This name seems inconsistent with the protoc-gen-alias path; does this work?

File: cmd/protoc-gen-alias/main.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+func generateFile(gen *protogen.Plugin, file *protogen.File) {
+	ourVersion := filepath.Base(filepath.Dir(file.Desc.Path()))
+	var versions []string
+	for _, msg := range file.Messages {
+		for _, line := range strings.Split(msg.Comments.Leading.String(), "\n") {
+			if strings.HasPrefix(line, "// +cue-gen:") {
+				items := strings.Split(line, ":")
+				if items[2] == "versions" {
```
Comment (keithmattix) [inline cmd/protoc-gen-alias/main.go:44]: nit: add a comment showing what a full, correctly parsed line should look like

Comment (keithmattix) [inline cmd/protoc-gen-alias/main.go:44]: Should we guard against an index out of bounds panic?

File: cmd/protoc-gen-alias/main.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+					for _, v := range strings.Split(items[3], ",") {
```
Comment (keithmattix) [inline cmd/protoc-gen-alias/main.go:45]: same here about index out of guarding; maybe just exist early if strings.Split returns too short a length

**Review by keithmattix — COMMENTED** [review]
Honestly, I'm not totally sure what's going on; the README confused me a bit. Could you clarify/update it?

File: cmd/protoc-gen-alias/README.md (hunk 1 above)
Comment (howardjohn) [reply cmd/protoc-gen-alias/README.md:12]: Sorry I copy+pasted another plugin and probably forgot the readme entirely. let me update

File: cmd/protoc-gen-alias/main.go (hunk 2 above)
Comment (howardjohn) [reply cmd/protoc-gen-alias/main.go:44]: I will add it, though fwiw the end result is largely the same (though we can have a nicer error)

**Review by keithmattix — APPROVED** [review]

**Review by dhawton — APPROVED** [review]

**Pull request merged**